### PR TITLE
fix(druid): Delete obsolete Druid NoSQL slice parameters

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

This PR addresses an issue raised by @padbk in https://github.com/apache/superset/discussions/24581#discussioncomment-6425918 related to sending an alert/revert via a CSV. The actual error was rather cryptic:

```
Error: Failed generating csv HTTP Error 400: BAD REQUEST
```

Specifically the RESTful `/api/v1/chart/{pk}/data/?format=csv` API endpoint was failing with the following error, 

```
{
  "message": "Request is incorrect: {'queries': {0: {'extras': {'having_druid': ['Unknown field.']}}}}"
}
```

which turned out to be related to https://github.com/apache/superset/pull/23997, i.e., the referenced PR (which I authored) didn't remove the now obsolete parameters from the `slices` table. This PR remedies said issue.

I used Airbnb's vast corpus (consisting of ~ 150k charts) to identify if/where the following form-data key were persisted to the database:

- `druid_time_origin`
- `having_druid`
- `having_filters`
- `granularity`

I wasn't able to find `having_filters` anywhere, but it was somewhat 😢 to see that the `druid_time_origin` and `granularity` fields were actually present in three separate places:

1. In the `slices.params` column
2. In the `form_data` field of the `slices.query_context` column.
3. In the `queries` field of the `slices.query_context` column.

### TESTING INSTRUCTIONS

CI and added/tested the DB migration. After running the migration I then confirmed that that I was successfully able to generate a report sent as a CSV.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [x] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [x] Migration is atomic, supports rollback & is backwards-compatible
  - [x] Confirm DB migration upgrade and downgrade tested
  - [x] Runtime estimates and downtime expectations provided << 5 minutes at Airbnb with ~ 150k charts
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/migrations/versions/2023-07-18_15-30_863adcf72773_delete_obsolete_druid_nosql_slice_parameters.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+                for key in ["druid_time_origin", "granularity"]:
+                    if key in params:
+                        del params[key]
+                        updated = True
+
+                if updated:
+                    slc.params = json.dumps(params)
+            except Exception:
+                pass
```
Comment (hughhhh) [inline superset/migrations/versions/2023-07-18_15-30_863adcf72773_delete_obsolete_druid_nosql_slice_parameters.py:67]: can we add a log here to see the exception

File: superset/migrations/versions/2023-07-18_15-30_863adcf72773_delete_obsolete_druid_nosql_slice_parameters.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+                    if extras := query.get("extras"):
+                        if "having_druid" in extras:
+                            del extras["having_druid"]
+                            updated = True
+
+                if updated:
+                    slc.query_context = json.dumps(query_context)
+            except Exception:
+                pass
```
Comment (hughhhh) [inline superset/migrations/versions/2023-07-18_15-30_863adcf72773_delete_obsolete_druid_nosql_slice_parameters.py:95]: see as comment above

File: superset/migrations/versions/2023-07-18_15-30_863adcf72773_delete_obsolete_druid_nosql_slice_parameters.py (hunk 1 above)
Comment (john-bodley) [reply superset/migrations/versions/2023-07-18_15-30_863adcf72773_delete_obsolete_druid_nosql_slice_parameters.py:67]: @hughhhh good idea.

**Comment by john-bodley** [comment]
@hughhhh I've addressed your comments. The exceptions are logged in the form,

```
ERROR [root] Unable to parse query context for slice 397732
Traceback (most recent call last):
  File "/srv/superset-internal/apache-superset/superset/migrations/versions/abc.py", line 75, in upgrade
    query_context = json.loads(slc.query_context)
  File "/usr/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.9/json/decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Unterminated string starting at: line 1 column 53523 (char 53522)

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Pull request merged**